File: Python/Practica 02 - toupper and tolower.py
def obtener_numero_int(valor):
    while True:
        entrada = input(valor)
        if entrada.replace(" ", "").isdigit():
            numero = int(entrada)
            return numero
        else:
            print("\n¡Error! Ingresa una opción correcta.\n")

def convertir_a_minuscula(cadena):
    resultado = ""
    for caracter in cadena:
        if 'A' <= caracter <= 'Z':
            resultado += chr(ord(caracter) + 32)
        elif caracter == 'Ñ':
            resultado += 'ñ'
        elif caracter == 'Á':
            resultado += 'á'
        elif caracter == 'É':
            resultado += 'é'
        elif caracter == 'Í':
            resultado += 'í'
        elif caracter == 'Ó':
            resultado += 'ó'
        elif caracter == 'Ú':
            resultado += 'ú'
        elif caracter == 'Ä':
            resultado += 'ä'
        elif caracter == 'Ë':
            resultado += 'ë'
        elif caracter == 'Ï':
            resultado += 'ï'
        elif caracter == 'Ö':
            resultado += 'ö'
        elif caracter == 'Ü':
            resultado += 'ü'
        else:
            resultado += caracter
    return resultado

def convertir_a_mayuscula(cadena):
    resultado = ""
    for caracter in cadena:
        if 'a' <= caracter <= 'z':
            resultado += chr(ord(caracter) - 32)
        elif caracter == 'ñ':
            resultado += 'Ñ'
        elif caracter == 'á':
            resultado += 'Á'
        elif caracter == 'é':
            resultado += 'É'
        elif caracter == 'í':
            resultado += 'Í'
        elif caracter == 'ó':
            resultado += 'Ó'
        elif caracter == 'ú':
            resultado += 'Ú'
        elif caracter == 'ä':
            resultado += 'Ä'
        elif caracter == 'ë':
            resultado += 'Ë'
        elif caracter == 'ï':
            resultado += 'Ï'
        elif caracter == 'ö':
            resultado += 'Ö'
        elif caracter == 'ü':
            resultado += 'Ü'
        else:
            resultado += caracter
    return resultado

opcion = 0

while opcion != 3:

    cadena = input("Ingresa tu cadena: ")
    print()
    print("1. Convertir a mayúsculas")
    print("2. Convertir a minúsculas")
    print("3. Salir")
    print()
    opcion = obtener_numero_int("Elija una opción: ")

    if opcion == 1:
        
        resultado = convertir_a_mayuscula(cadena)
        print(f"Tu cadena {cadena} se convirtió en: {resultado}")
        print()

    elif opcion == 2:

        resultado = convertir_a_minuscula(cadena)
        print(f"Tu cadena {cadena} se convirtió en: {resultado}")
        print()

    elif opcion == 3:

        print("Saliste del programa")

File: C/Practica 02 - toupper and tolower.c
#include <stdio.h>
void convertir_a_Minuscula(char *cadena)
{
    while (*cadena)
    {
        if (*cadena >= 65 && *cadena <= 90)
        {
            *cadena = *cadena + 32;
        }
        else if(*cadena == 165)//Ñ
        {
            *cadena = *cadena - 1;
        }
        else if(*cadena == 181)//Á
        {
            *cadena = *cadena - 21;
        }
        else if(*cadena == 144)//É
        {
            *cadena = *cadena - 14;
        }
        else if(*cadena == 214)//Í
        {
            *cadena = *cadena - 53;
        }
        else if(*cadena == 224)//Ó
        {
            *cadena = *cadena - 62;
        }
        else if(*cadena == 233)//Ú
        {
            *cadena = *cadena - 70;
        }
        else if(*cadena == 142)//Ä
        {
            *cadena = *cadena - 10;
        }
        else if(*cadena == 211)//Ë
        {
            *cadena = *cadena - 74;
        }
        else if(*cadena == 216)//Ï
        {
            *cadena = *cadena - 77;
        }
        else if(*cadena == 153)//Ö
        {
            *cadena = *cadena - 5;
        }
        else if(*cadena == 154)//Ü
        {
            *cadena = *cadena - 25;
        }
        *cadena++;
    }
}
void convertir_a_Mayuscula(char *cadena)
{
    while (*cadena)
    {
        if (*cadena >= 97 && *cadena <= 122)
        {
            *cadena = *cadena - 32;
        }
        else if(*cadena == 164)//ñ
        {
            *cadena = *cadena + 1;
        }
        else if(*cadena == 160)//á
        {
            *cadena = *cadena + 21;
        }
        else if(*cadena == 130)//é
        {
            *cadena = *cadena + 14;
        }
        else if(*cadena == 161)//í
        {
            *cadena = *cadena + 53;
        }
        else if(*cadena == 162)//ó
        {
            *cadena = *cadena + 62;
        }
        else if(*cadena == 163)//ú
        {
            *cadena = *cadena + 70;
        }
        else if(*cadena == 132)//ä
        {
            *cadena = *cadena + 10;
        }
        else if(*cadena == 137)//ë
        {
            *cadena = *cadena + 74;
        }
        else if(*cadena == 139)//ï
        {
            *cadena = *cadena + 77;
        }
        else if(*cadena == 148)//ö
        {
            *cadena = *cadena + 5;
        }
        else if(*cadena == 129)//ü
        {
            *cadena = *cadena + 25;
        }
        *cadena++;
    }
}
int main()
{
    int opcion = 0;
    char texto[60];
    do
    {
        printf("Ingresa tu cadena: ");
        scanf("%s", texto);
